dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -333,7 +336,17 @@ private String readFileFromRepo(final String originalFileName, final String orig
             if (decodedContentAndMetadata == null) {
                 return null;
             } else {
-                return decodedContentAndMetadata.getRight();
+                String content = decodedContentAndMetadata.getRight();
+                String encoding = decodedContentAndMetadata.getLeft().getEncoding();
+                // If the file size is 1MB or larger, content will be "" and the encoding will be "none":
+                // https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28 (see "Notes")
+                // In such a case, we retrieve the content via the blob endpoint.
+                if ("".equals(content) && "none".equals(encoding)) {
```

Comment:
Has this been tested with completely empty files?

Reply:
Or will that go down `decodedContentAndMetadata == null` ?

Reply:
> Has this been tested with completely empty files?

I should add some tests for the empty and "excessively large" cases.

Reply:
> Or will that go down `decodedContentAndMetadata == null` ?

?

Reply:
The conditional statement before this section, it's possible that empty files are handled by that already. 
Just suggesting testing to make sure